param/parameterized.py

```diff
@@ -804,19 +817,19 @@ def inner(*args, **kwargs):
         inner.__doc__= "Inspect .param.%s method for the full docstring"  % fn.__name__
         return inner
 
-
     @classmethod
     def _changed(cls, change):
         """
         Predicate that determines whether a Change object has actually
         changed such that old!=new.
         """
         try:  # To be improved by adding better machinery to test equality for complex types
+            if not is_comparable(change.new) or not is_comparable(change.old):
+                return True
```

Comment:
Hmm.  Maybe instead of having a separate is_comparable test and a hope of adding better machinery, those two things should be combined so that this class has a class attribute that stores the mechanism used to decide if it's changed, which currently would use is_comparable + the bool() test, but which someone could replace with a more full-featured test?

Reply:
If we decide to implement a more full-featured approach wouldn't we be adding it to param anyway and replacing this code?

Reply:
Eventually, sure, but it would (a) help us develop it without having to tightly link the param version with other code versions (which has been painful over the past few weeks), and (b) this would allow other people who are doing this to develop and use their own such code if it came up.  My point is just that if we are already abstracting this, may as well go all the way and make it pluggable.

Reply:
I have no objection. I was just thinking that implementing a way to declare equality functions for different types as I suggested in https://github.com/ioam/param/issues/269 would not take too long to implement and would offer a pluggable mechanism.

Reply:
Also fine.